public/app/features/variables/pickers/OptionsPicker/actions.ts

```diff
@@ -88,9 +94,11 @@ export const commitChangesToVariable = (): ThunkResult<void> => {
       return;
     }
 
-    const adapter = variableAdapters.get(updated.type);
-    await adapter.setValue(updated, updated.current, true);
-    return;
+    if (callback) {
+      return callback(updated);
+    } else {
+      return setVariable(updated);
+    }
```

Comment:
Nit: maybe we could rewrite this as:
```suggestion
    if (callback) {
      return callback(updated);
    } 
    
    return setVariable(updated);
```